includes/codegen/controls/AbstractControl_CodeGenerator.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
```

Comment:
Why is this called AbstractControl_CodeGenerator and not ControlBase_CodeGenerator to correspond with the ControlBase file?